src/hotspot/share/opto/node.hpp

```diff
@@ -1628,6 +1635,32 @@ class Node_List : public Node_Array {
   void dump_simple() const;
 };
 
+// Definition must appear after complete type definition of Node_List
+template <typename Callback, typename Check>
+void Node::visit_uses(Callback callback, Check is_boundary) const {
+  ResourceMark rm;
+  VectorSet visited;
+  Node_List worklist;
+
+  // The initial worklist consists of the direct uses
+  for (DUIterator_Fast kmax, k = fast_outs(kmax); k < kmax; k++)
+    worklist.push(fast_out(k));
+
+  while (worklist.size() > 0) {
+    Node* use = worklist.pop();
+    // Skip already visited nodes
+    if (visited.test_set(use->_idx)) continue;
+    // Apply callback on boundary nodes
+    if (is_boundary(use)) callback(use);
```

Comment:
We are supposed to always use the block curly brackets.

Reply:
Thanks, fixed.